Sources/xcproj/PBXProjObjects+Helpers.swift

```diff
@@ -15,7 +16,224 @@ public extension PBXProj.Objects {
+    public func group(named groupName: String, inGroup: PBXGroup) -> (reference: String, group: PBXGroup)? {
+        let children = inGroup.children
+        return groups.first(where: {
+            children.contains($0.key) && ($0.value.name == groupName || $0.value.path == groupName)
+        }).map({ ($0.0, $0.1) })
+    }
+
+    /// Adds new group with the give name to the given parent group.
+    /// Group name can be a path with components separated by `/`.
+    /// Will create new groups for intermediate path components or use existing groups.
+    /// Returns all new or existing groups in the path and their references.
+    ///
+    /// - Parameters:
+    ///   - groupName: group name, can be a path with components separated by `/`
+    ///   - toGroup: parent group
+    ///   - options: additional options, default is empty set.
+    /// - Returns: all new or existing groups in the path and their references.
+    public func addGroup(named groupName: String, to toGroup: PBXGroup, options: GroupAddingOptions = []) -> [(reference: String, group: PBXGroup)] {
+        return addGroups(groupName.components(separatedBy: "/"), to: toGroup, options: options)
+    }
+
+    private func addGroups(_ groupNames: [String], to toGroup: PBXGroup, options: GroupAddingOptions) -> [(reference: String, group: PBXGroup)] {
+        guard !groupNames.isEmpty else { return [] }
+        let newGroup = createOrGetGroup(named: groupNames[0], in: toGroup, options: options)
+        return [newGroup] + addGroups(Array(groupNames.dropFirst()), to: newGroup.1, options: options)
+    }
+
+    private func createOrGetGroup(named groupName: String, in parentGroup: PBXGroup, options: GroupAddingOptions) -> (reference: String, group: PBXGroup) {
+        if let existingGroup = group(named: groupName, inGroup: parentGroup) {
+            return existingGroup
+        }
+
+        let newGroup = PBXGroup(
+            children: [],
+            sourceTree: .group,
+            name: groupName,
+            path: options.contains(.withoutFolder) ? nil : groupName
+        )
+        let reference = generateReference(newGroup, groupName)
+        addObject(newGroup, reference: reference)
+        parentGroup.children.append(reference)
+        return (reference, newGroup)
+    }
+
+    /// Adds file at the give path to the project and given group or returns existing file and its reference.
+    ///
+    /// - Parameters:
+    ///   - filePath: path to the file.
+    ///   - toGroup: group to add file to.
+    ///   - sourceTree: file sourceTree, default is `.group`
+    ///   - sourceRoot: path to project's source root.
+    /// - Returns: new or existing file and its reference.
+    public func addFile(at filePath: Path, toGroup: PBXGroup, sourceTree: PBXSourceTree = .group, sourceRoot: Path) throws -> (reference: String, file: PBXFileReference) {
+        guard filePath.isFile else {
```

Comment:
Why is it necessary to check if the file really exists?

Reply:
First to not make project buildable otherwise (it will not build if this file reference is then added to the build phase). Second to not allow paths to folders.

Reply:
Actually you can non existing add file to build phase. This is common case with generated files. But as this is a helper function, I believe it is ok to make these kinds of checks here.